File: services/xena-service-atila/app/Domains/Message.ts
import { validString, validEnum } from './Validators'
import { v4 as uuidv4 } from 'uuid'

type MessageStatus = 'SEEN' | 'SENT'

export default class Message {
  constructor (
    public readonly id: string,
    public readonly from: string | null,
    public readonly to: string | null,
    public readonly subject: string,
    public readonly content: string,
    public readonly status: MessageStatus,
    public readonly replyTo: string | null,
  ) {
    this.id = validString(id ?? uuidv4(), 'BAD_MESSAGE_ID', 'NON_EMPTY')
    this.from = from ? validString(from, 'BAD_MESSAGE_FROM', 'NON_EMPTY') : null
    this.to = to ? validString(to, 'BAD_MESSAGE_TO', 'NON_EMPTY') : null
    this.subject = validString(subject, 'BAD_MESSAGE_SUBJECT', 'NON_EMPTY')
    this.content = validString(content, 'BAD_MESSAGE_CONTENT', 'NON_EMPTY')
    this.status = validEnum(status, ['SEEN', 'SENT'], 'BAD_MESSAGE_STATUS')
    this.replyTo = replyTo ? validString(replyTo, 'BAD_MESSAGE_REPLY_TO_ID', 'NON_EMPTY') : null
  }

  public static fromJSON = (json) => new Message(
      json.id,
      json.from,
      json.to,
      json.subject,
      json.content,
      json.status,
      json.replyTo,
    )
}

File: bots/xena-bot-anaconda/domains/message.py
from domains.validators import Validators

class Message:
  def __init__(self, id: str, from_id: str, to: str, subject: str, content: str, status: str, reply_to: str,):
    self.id = Validators.valid_string(id, 'BAD_MESSAGE_ID')
    self.from_id = from_id
    # self.from_id = Validators.valid_string(from_id, 'BAD_MESSAGE_FROM_ID')
    self.to = Validators.valid_string(to, 'BAD_MESSAGE_TO')
    self.subject = Validators.valid_string(subject, 'BAD_MESSAGE_SUBJECT')
    self.content = Validators.valid_string(content, 'BAD_MESSAGE_CONTENT')
    self.status = Validators.valid_string(status, 'BAD_MESSAGE_STATUS')
    self.reply_to = reply_to
    # self.reply_to = Validators.valid_string(reply_to, 'BAD_MESSAGE_REPLY_TO')

  @staticmethod
  def from_json(json):
    return Message(
      id = json['id'],
      from_id = json['from'],
      to = json['to'],
      subject = json['subject'],
      content = json['content'],
      status = json['status'],
      reply_to = json['replyTo'],
    )

  def serialize(self):
    return {
      'id': self.id,
      'from': self.from_id,
      'to': self.to,
      'subject': self.subject,
      'content': self.content,
      'status': self.status,
      'replyTo': self.reply_to,
    }